src/AspNet.Security.OAuth.Alipay/AlipayAuthenticationHandler.cs

```diff
@@ -0,0 +1,242 @@
+            [NotNull] AuthenticationProperties properties,
+            [NotNull] string redirectUri)
+        {
+            string stateValue = Options.StateDataFormat.Protect(properties);
+
+            string address = QueryHelpers.AddQueryString(Options.AuthorizationEndpoint, new Dictionary<string, string>
+            {
+                ["app_id"] = Options.ClientId,
+                ["redirect_uri"] = redirectUri,
+                ["scope"] = FormatScope(),
+                ["state"] = stateValue
+            });
+
+            return address;
+        }
+
+        protected override async Task<HandleRequestResult> HandleRemoteAuthenticateAsync()
+        {
+            var query = Request.Query.ToDictionary(c => c.Key, c => c.Value, StringComparer.OrdinalIgnoreCase);
+
+            if (query.TryGetValue("auth_code", out var authCode))
+            {
+                query["code"] = authCode;
+                Request.QueryString = QueryString.Create(query);
+            }
+
+            return await base.HandleRemoteAuthenticateAsync();
+        }
+
+        protected override async Task<OAuthTokenResponse> ExchangeCodeAsync([NotNull] OAuthCodeExchangeContext context)
+        {
+            var sortedParams = new SortedDictionary<string, string>()
+            {
+                ["app_id"] = Options.ClientId,
+                ["charset"] = "UTF-8",
+                ["code"] = context.Code,
+                ["format"] = "JSON",
+                ["grant_type"] = "authorization_code",
+                ["method"] = "alipay.system.oauth.token",
+                ["sign_type"] = "RSA2",
+                ["timestamp"] = Clock.UtcNow.ToString("yyyy-MM-dd HH:mm:ss", new CultureInfo("zh-CN")),
+                ["version"] = "1.0",
+            };
+            sortedParams.Add("sign", GetRSA2Signature(sortedParams, Options.ClientSecret));
+
+            string address = QueryHelpers.AddQueryString(Options.TokenEndpoint, sortedParams);
+
+            using var request = new HttpRequestMessage(HttpMethod.Get, address);
+            using var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving an access token: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                return OAuthTokenResponse.Failed(new Exception("An error occurred while retrieving an access token."));
+            }
+
+            using var payload = JsonDocument.Parse(await GetRawStringAsync(response, "alipay_system_oauth_token_response"));
```

Comment:
```suggestion
            var payload = JsonDocument.Parse(await GetRawStringAsync(response, "alipay_system_oauth_token_response"));
```

Reply:
Control of the `JsonDocument` is transferred to the `OAuthTokenResponse` instance returned (which implements `IDisposable`) in the case of `ExchangeCodeAsync()`, so shouldn't use a `using`.